controllers/topology/internal/scope/upgradetracker.go

```diff
@@ -22,31 +22,68 @@ const maxMachineDeploymentUpgradeConcurrency = 1
 
 // UpgradeTracker is a helper to capture the upgrade status and make upgrade decisions.
 type UpgradeTracker struct {
+	ControlPlane       ControlPlaneUpgradeTracker
 	MachineDeployments MachineDeploymentUpgradeTracker
 }
 
+// ControlPlaneUpgradeTracker holds the current upgrade status of the Control Plane.
+type ControlPlaneUpgradeTracker struct {
+	IsAtTopologyVersion bool
```

Comment:
This seems equivalent to HasPendingUpdates for machine deployments.
Can we converge on a consistent naming?